test/integration/fixtures.test.js

```diff
@@ -58,30 +58,36 @@ test('fixtures', (done) => {
     const fixture = fixtureWithName.fixture;
 
     uploadFixture(fixtureName, (err, parts) => {
-      if (err) {
-        err.fixtureName = fixtureName;
-        throw err;
-      }
+      try {
+        if (err) {
+          err.fixtureName = fixtureName;
+          throw err;
+        }
 
-      fixture.forEach((expectedPart, i) => {
-        const parsedPart = parts[i];
-        strictEqual(parsedPart.type, expectedPart.type);
-        strictEqual(parsedPart.name, expectedPart.name);
-
-        if (parsedPart.type === 'file') {
-          const file = parsedPart.value;
-          strictEqual(file.originalFilename, expectedPart.originalFilename,
-            `${JSON.stringify([expectedPart, file])}`);
-
-          if (expectedPart.sha1) {
-            strictEqual(
-              file.hash,
-              expectedPart.sha1,
-              `SHA1 error ${file.originalFilename} on ${file.filepath} ${JSON.stringify([expectedPart, file])}`,
-            );
+        fixture.forEach((expectedPart, i) => {
+          const parsedPart = parts[i];
+          strictEqual(parsedPart.type, expectedPart.type);
+          strictEqual(parsedPart.name, expectedPart.name);
+
+          if (parsedPart.type === 'file') {
+            const file = parsedPart.value;
+            strictEqual(file.originalFilename, expectedPart.originalFilename,
+              `${JSON.stringify([expectedPart, file])}`);
+
+            if (expectedPart.sha1) {
+              strictEqual(
+                file.hash,
+                expectedPart.sha1,
+                `SHA1 error ${file.originalFilename} on ${file.filepath} ${JSON.stringify([expectedPart, file])}`,
+              );
+            }
           }
-        }
-      });
+        });
+      } catch (e) {
+        server.close();
+        done(e);
+        throw e;
```

Comment:
This new try-catch is because otherwise the test was just timing out rather than reporting the error.